controllers/openstackcluster_controller.go

```diff
@@ -585,6 +585,62 @@ func bastionHashHasChanged(computeHash string, clusterAnnotations map[string]str
 	return latestHash != computeHash
 }
 
+func reconcileLoadBalancerNetwork(openStackCluster *infrav1.OpenStackCluster, networkingService *networking.Service) error {
+	if openStackCluster.Spec.APIServerLoadBalancer.Network != nil {
+		lbNetOpts := openStackCluster.Spec.APIServerLoadBalancer.Network.ToListOpt()
+		lbNetList, err := networkingService.GetNetworksByFilter(&lbNetOpts)
+		if err != nil {
+			handleUpdateOSCError(openStackCluster, fmt.Errorf("failed to find loadbalancer network: %w", err))
+			return fmt.Errorf("failed to find network: %w", err)
+		}
+		if len(lbNetList) == 0 {
+			handleUpdateOSCError(openStackCluster, fmt.Errorf("failed to find ana loadbalancer network"))
+			return fmt.Errorf("failed to find any network")
+		}
+		if len(lbNetList) > 1 {
+			handleUpdateOSCError(openStackCluster, fmt.Errorf("found multiple loadbalancer networks (result: %v)", lbNetList))
+			return fmt.Errorf("found multiple loadbalancer networks (result: %v)", lbNetList)
+		}
+
+		// Filter out only relevant subnets specified by the spec
+		if openStackCluster.Status.APIServerLoadBalancer == nil {
+			openStackCluster.Status.APIServerLoadBalancer = &infrav1.LoadBalancer{
+				LoadBalancerNetwork: &infrav1.NetworkStatusWithSubnets{
+					NetworkStatus: infrav1.NetworkStatus{
+						Name: lbNetList[0].Name,
+						ID:   lbNetList[0].ID,
+					},
+				},
+			}
+		}
+		openStackCluster.Status.APIServerLoadBalancer.LoadBalancerNetwork.Subnets = []infrav1.Subnet{}
+		for _, subnetID := range lbNetList[0].Subnets {
+			for _, s := range openStackCluster.Spec.APIServerLoadBalancer.Subnets {
+				if subnetID == s.ID {
+					openStackCluster.Status.APIServerLoadBalancer.LoadBalancerNetwork.Subnets = append(
+						openStackCluster.Status.APIServerLoadBalancer.LoadBalancerNetwork.Subnets, infrav1.Subnet{
+							ID:   s.ID,
+							Name: s.Name,
+							CIDR: s.CIDR,
+						})
+				}
+			}
+
+			// if no match was found, add all subnets within openStackCluster.Status.APIServerLoadBalancer.Network
+			if len(openStackCluster.Status.APIServerLoadBalancer.LoadBalancerNetwork.Subnets) == 0 {
+				for _, subnetID := range lbNetList[0].Subnets {
+					openStackCluster.Status.APIServerLoadBalancer.LoadBalancerNetwork.Subnets = append(
+						openStackCluster.Status.APIServerLoadBalancer.LoadBalancerNetwork.Subnets, infrav1.Subnet{
+							ID: subnetID,
+						})
+				}
+			}
```

Comment:
This behavior feels counter-intuitive. If users specified wrong subnet IDs, we'll silently ignore their mistake. We should rather explicitly fail here.

Also what if 2 subnets were specified and we only found 1? IMO that should be an error too.

Reply:
gotcha.. thanks